matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/state/DefaultStateService.kt

```diff
@@ -84,4 +101,67 @@ internal class DefaultStateService @AssistedInject constructor(@Assisted private
                 stateKey = null
         )
     }
+
+    override fun updateName(name: String, callback: MatrixCallback<Unit>): Cancelable {
+        return sendStateEvent(
+                eventType = EventType.STATE_ROOM_NAME,
+                body = mapOf("name" to name),
+                callback = callback,
+                stateKey = null
+        )
+    }
+
+    override fun updateCanonicalAlias(alias: String, callback: MatrixCallback<Unit>): Cancelable {
+        return sendStateEvent(
+                eventType = EventType.STATE_ROOM_CANONICAL_ALIAS,
+                body = mapOf("alias" to alias),
+                callback = callback,
+                stateKey = null
+        )
+    }
+
+    override fun updateHistoryReadability(readability: String, callback: MatrixCallback<Unit>): Cancelable {
+        return sendStateEvent(
+                eventType = EventType.STATE_ROOM_HISTORY_VISIBILITY,
+                body = mapOf("history_visibility" to readability),
+                callback = callback,
+                stateKey = null
+        )
+    }
+
+    override fun updateAvatar(avatarUri: Uri, fileName: String, callback: MatrixCallback<Unit>): Cancelable {
+        val cancelableBag = CancelableBag()
+        val workerParams = UploadAvatarWorker.Params(sessionId, avatarUri, fileName)
+        val workerData = WorkerParamsFactory.toData(workerParams)
+
+        val uploadAvatarWork = workManagerProvider.matrixOneTimeWorkRequestBuilder<UploadAvatarWorker>()
```

Comment:
Same remark than the other PR with uploading avatar, don't think we should use a worker overthere